src/coordination/data_instance_management_server_handlers.cpp

```diff
@@ -79,19 +79,21 @@ void DataInstanceManagementServerHandlers::StateCheckHandler(replication::Replic
   coordination::StateCheckReq req;
   slk::Load(&req, req_reader);
 
-  bool const is_replica = replication_handler.IsReplica();
-  auto const uuid = std::invoke([&replication_handler, is_replica]() -> std::optional<utils::UUID> {
+  auto locked_repl_state = replication_handler.GetReplState();
```

Comment:
I think you can use here read lock.

is_replica is read only
uuid is read only
writing_enabled is read only